x-pack/plugins/cases/server/services/notifications/email_notification_service.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IBasePath, Logger } from '@kbn/core/server';
+import type { NotificationsPluginStart } from '@kbn/notifications-plugin/server';
+import type { SecurityPluginStart } from '@kbn/security-plugin/server';
+import type { UserProfileUserInfo } from '@kbn/user-profile-components';
+import { CASE_SAVED_OBJECT } from '../../../common/constants';
+import type { CaseSavedObject } from '../../common/types';
+import { getCaseViewPath } from '../../common/utils';
+import type { NotificationService, NotifyArgs } from './types';
+
+type WithRequiredProperty<T, K extends keyof T> = T & Required<Pick<T, K>>;
+
+type UserProfileUserInfoWithEmail = WithRequiredProperty<UserProfileUserInfo, 'email'>;
+
+export class EmailNotificationService implements NotificationService {
+  private readonly logger: Logger;
+  private readonly notifications: NotificationsPluginStart;
+  private readonly security: SecurityPluginStart;
+  private readonly publicBaseUrl?: IBasePath['publicBaseUrl'];
+
+  constructor({
+    logger,
+    notifications,
+    security,
+    publicBaseUrl,
+  }: {
+    logger: Logger;
+    notifications: NotificationsPluginStart;
+    security: SecurityPluginStart;
+    publicBaseUrl?: IBasePath['publicBaseUrl'];
+  }) {
+    this.logger = logger;
+    this.notifications = notifications;
+    this.security = security;
+    this.publicBaseUrl = publicBaseUrl;
+  }
+
+  private getTitle(theCase: CaseSavedObject) {
+    return `[Elastic] ${theCase.attributes.title}`;
+  }
+
+  private getMessage(theCase: CaseSavedObject) {
+    const lineBreak = '\r\n\r\n';
+    let message = `You got assigned to an Elastic Case.${lineBreak}`;
+    message = `${message}Title: ${theCase.attributes.title}${lineBreak}`;
+    message = `${message}Status: ${theCase.attributes.status}${lineBreak}`;
+    message = `${message}Severity: ${theCase.attributes.severity}${lineBreak}`;
+    message = `${message}Tags: ${theCase.attributes.tags.join(',')}${lineBreak}`;
+
+    if (this.publicBaseUrl) {
+      const caseUrl = getCaseViewPath({
+        publicBaseUrl: this.publicBaseUrl,
+        caseId: theCase.id,
+        owner: theCase.attributes.owner,
+      });
+
+      message = `${message}${lineBreak}[View case](${caseUrl})`;
+    }
+
+    return message;
+  }
+
+  public async notifyAssignees({ assignees, theCase }: NotifyArgs) {
+    if (!this.notifications.isEmailServiceAvailable()) {
```

Comment:
Don't know if it's possible for `isEmailServiceAvailable()` to throw an error but maybe we should move it into the `try`?